changelog/range-error.md

```diff
@@ -0,0 +1,54 @@
+Out of bounds array access now gives a fluent error message
+
+Errors resulting from bad indexing will now contain the length of
+the array, as well as the offending index for `arr[bad]`,
+or offending indices `arr[bad1 .. bad2]` for bad slices.
+
+For example:
+
+---
+void main()
+{
+    int[] a = [1, 2, 3];
+    int b = a[7];
+}
+---
+
+Previously this would yield the following error when compiled and run:
+
+$(CONSOLE
+> dmd -run main.d
+core.exception.RangeError@main.d(4): Range violation
+$(NDASH)$(NDASH)$(NDASH)$(NDASH)$(NDASH)$(NDASH)$(NDASH)$(NDASH)$(NDASH)$(NDASH)
+??:? _d_arrayboundsp [0x555765c167f9]
+??:? _Dmain [0x555765c16752]
+)
```

Comment:
The changelog doesn't mention the two new classes derived from `Error`, except for the implicit mentions in the code sample. Not sure if you want to add it here or not.

Reply:
I mentioned them at first, but I removed it because I don't want to encourage D users to catch range errors.